### ISSUE TITLE
RuntimeError: DDtrace is not compatible with OpenAI `with_raw_response`

### ISSUE THREAD

**Issue by jsullivan-zonos** [DataDog/dd-trace-py#7865]
### Summary of problem
An exception raised by ddtrace when using openai's `with_raw_response`. This does not happen when omitting the `.with_raw_response`.

**Mininal test case**
```python
import openai

client = openai.OpenAI()

chat_completion = client.chat.completions.with_raw_response.create(
    messages=[
        {
            "role": "user",
            "content": "Say this is a test",
        }
    ],
    model="gpt-3.5-turbo",
)
```

**Traceback**
```
$ ddtrace-run python test_ddtrace_openai.py 
Error submitting packet: [Errno 111] Connection refused, dropping the packet and closing the socket
Traceback (most recent call last):
  File "/home/jonathan/test_ddtrace_openai.py", line 5, in <module>
    chat_completion = client.chat.completions.with_raw_response.create(
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/openai/_response.py", line 244, in wrapped
    return cast(APIResponse[R], func(*args, **kwargs))
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/openai/_utils/_utils.py", line 271, in wrapper
    @functools.wraps(func)
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py", line 411, in patched_endpoint
    g.send((resp, err))
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/ddtrace/contrib/openai/patch.py", line 376, in _traced_endpoint
    hook.send((resp, err))
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/ddtrace/contrib/openai/_endpoint_hooks.py", line 71, in handle_request
    return self._record_response(pin, integration, span, args, kwargs, resp, error)
  File "/home/jonathan/miniconda3/lib/python3.9/site-packages/ddtrace/contrib/openai/_endpoint_hooks.py", line 275, in _record_response
    choices = resp.choices or []
AttributeError: 'APIResponse' object has no attribute 'choices'
failed to send, dropping 1 traces to intake at http://localhost:8126/v0.4/traces after 3 retries
```

### Which version of dd-trace-py are you using?
2.3.1

### Which version of pip are you using?
23.0.1

### Which libraries and their versions are you using?
openai==1.3.7

<details>
  <summary>`pip freeze`</summary>
  
</details>

### How can we reproduce your problem?
Make an OpenAI request using the `with_raw_response` modifier.

### What is the result that you get?
An exception is raised when trying to record the OpenAI usage, halting the instrumented program.

### What is the result that you expected?
No exception being raised and proper tracing on OpenAI spans.

**Comment by emmettbutler** [comment]
@jsullivan-zonos thank you for the issue report! What you've posted looks like a partial traceback - is there more to it? If so, it would be helpful to see the full stacktrace including the exception type and originating line of code.

**Comment by emmettbutler** [comment]
@Yun-Kim

**Comment by jsullivan-zonos** [comment]
I've just updated the issue with a minimum test case.

**Comment by Yun-Kim** [comment]
Hi @jsullivan-zonos, thanks for reaching out! 

We've already addressed this in #7733, but hasn't been released yet. I've confirmed that this indeed isn't an issue with the latest main version of the tracer, so please try again once 2.3.2 or 2.2.2 are released sometime in the next two weeks. I'll close this for now, but feel free to reopen if you still see any issues then.
